content/blog/2020-07-22-july-20-community-gems.md

```diff
@@ -0,0 +1,169 @@
+
+The first release of CML is compatible with GitHub and GitLab. We've seen
+[many requests for Bitbucket support](https://github.com/iterative/cml/issues/140),
+and we're making it a development priority for an upcoming release. Stay tuned.
+
+### [Q: I have on-premise GPUs. Can CML use them to execute pipelines?](https://discordapp.com/channels/485586884165107732/728693131557732403/730070747388706867)
+
+Yep! You can use on-premise compute resources by configuring them as self-hosted
+runners. See
+[GitHub](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners)
+and [GitLab](https://docs.gitlab.com/runner/)'s official docs for more details
+and setup instructions.
+
+### [Q: I'm building a workflow that deploys a GCP Compute Engine instance, but I can only find examples with AWS EC2 in the CML docs. What do I do?](https://discordapp.com/channels/485586884165107732/728693131557732403/730688592787275806)
+
+There is a slight difference in the way CML handles credentials for AWS and GCP,
+and that means you'll have to modify your workflow file slightly. We've added
+this example to our
+[project README](https://github.com/iterative/cml#allocating-cloud-resources-with-cml):
+
+```yaml
+deploy-gce:
+  runs-on: [ubuntu-latest]
+  container: docker://dvcorg/cml-cloud-runner
+
+  steps:
+    - name: deploy
+      shell: bash
+      env:
+        repo_token: ${{ secrets.REPO_TOKEN }}
+        GOOGLE_APPLICATION_CREDENTIALS_DATA:
+          ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS_DATA }}
+      run: |
+
+        echo '${{ secrets.GOOGLE_APPLICATION_CREDENTIALS_DATA }}' > gce-credentials.json
+        export GOOGLE_APPLICATION_CREDENTIALS='gce-credentials.json'
+
+        RUNNER_LABELS="gce"
+        RUNNER_REPO="https://github.com/${GITHUB_REPOSITORY}"
+        MACHINE="cml$(date +%s)"
+
+        docker-machine create --driver google \
+          --google-machine-type <machine-type> \
+          --google-project <project-id> \
+          $MACHINE
+
+        eval "$(docker-machine env --shell sh $MACHINE)"
+
+        (
+        docker-machine ssh $MACHINE "sudo mkdir -p /docker_machine && sudo chmod 777 /docker_machine" && \
+        docker-machine scp -r -q ~/.docker/machine/ $MACHINE:/docker_machine && \
+        docker-machine scp -q gce-credentials.json $MACHINE:/docker_machine/gce-credentials.json && \
+
+        eval "$(docker-machine env --shell sh $MACHINE)" && \
+        docker run --name runner -d \
+          -v /docker_machine/gce-credentials.json:/gce-credentials.json \
+          -e GOOGLE_APPLICATION_CREDENTIALS='/gce-credentials.json' \
+          -v /docker_machine/machine:/root/.docker/machine \
+          -e DOCKER_MACHINE=$MACHINE \
+          -e repo_token=$repo_token \
+          -e RUNNER_LABELS=$RUNNER_LABELS \
+          -e RUNNER_REPO=$RUNNER_REPO \
+          -e RUNNER_IDLE_TIMEOUT=120 \
+          dvcorg/cml-cloud-runner:latest && \
+        sleep 20 && echo "Deployed $MACHINE"
```

Comment:
curious - why do we need sleep 20 here?

Reply:
it's arbitrary how long, but not sure why we picked 20- question for David